### PR TITLE
Prevent literal strings in component source code

### LINKED ISSUE #2184: Prevent text literals in JSX
Enable
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-literals.md

to prevent direct strings. This is because all text should be able to be changed by the consumer. See https://github.com/salesforce/design-system-react/blob/master/docs/codebase-overview.md#rendered-text-needs-a-prop-group-assistive-text-and-labels

### PR DESCRIPTION
**Pull request by interactivellama**
Fixes #2184

### Additional description
* Adds `assistiveText` and `labels` strings
* Wraps non-words in `{ }`

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [ ] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [ ] `npm run lint:fix` has been run and linting passes.
* [ ] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

**Comment by interactivellama** [comment]
@pulkonet @aswinshenoy 

Per https://github.com/salesforce/design-system-react/blob/master/docs/codebase-overview.md#rendered-text-needs-a-prop-group-assistive-text-and-labels all text should be able to be changed by the consumer.

`WelcomeMat`, `TrialBar`, `ProgressBar` had text in them, and we missed it in code reviews.

After this ESlint rule is added, strings as JSX children should get flagged. `{'* '}` is totally cool for non-words though.

File: components/modal/private/manager.jsx [hunk 1]
```diff
@@ -95,6 +95,8 @@ class Manager extends React.Component {
 						onClick={this.closeModal}
 					>
 						<Icon name="close" category="utility" size="small" />
+						{/* This file is an example file and should not be used */}
+						{/* eslint-disable-next-line react/jsx-no-literals */}
```
Comment (kevinparkerson) [inline components/modal/private/manager.jsx]: could this file be deleted instead?

Comment (interactivellama) [reply components/modal/private/manager.jsx]: I'd prefer a separate PR.

Comment (interactivellama) [reply components/modal/private/manager.jsx]: So we can call it out on the release notes

Comment (interactivellama) [reply components/modal/private/manager.jsx]: There's a lot of cruft in the library. https://github.com/salesforce/design-system-react/blob/master/components/modal/index.jsx#L226

Thankfully, the library is mostly functional programming, so we haven't had a lot of bugs.

Comment (interactivellama) [reply components/modal/private/manager.jsx]: @kevinparkerson #2189 added for this issue

**Pull request merged**